src/adapter/src/rbac.rs

```diff
@@ -137,6 +137,47 @@ pub fn check_command(catalog: &Catalog, cmd: &Command) -> Result<(), Unauthorize
     }
 }
 
+/// Checks if a session is authorized to purify a CREATE SOURCE statement. Usually authorization is
+/// checked after planning by [`check_plan`]. However CREATE SOURCE statements are purified
+/// before planning, which may require the use of some connections and secrets.
+pub fn check_purify_create_source(
```

Comment:
This function looks like it could be general for any pre-plan check. Does it make sense to name it a more general thing?

Also, does it make sense to have `check_plan` call this function too, instead of duplicating any code? As in: there's some rbac checking we can only do pre-plan, but even if we have a plan we still want to run that code.

Reply:
So I converted this into `check_item_usage` so that it's general for checking `USAGE` privileges on all items. Then this functions is called by `check_plan`, I removed all item `USAGE` from `check_plan`. Let me know what you think of that approach.